src/main/appender.cpp

```diff
@@ -373,6 +373,13 @@ void BaseAppender::Append(Value value) { // NOLINT: template stuff
 	AppendValue(value);
 }
 
+void BaseAppender::Append(DataChunk &target, const Value &value, idx_t row, idx_t column) {
+	if (column >= target.ColumnCount()) {
+		throw InvalidInputException("Too many appends for chunk!");
+	}
+	target.SetValue(column, row, value);
```

Comment:
Do you mean comparing row to the `chunk.size` ?

Reply:
I mean the number of rows that can be stored (maximum) in the data chunk. 
```cpp
	//! The amount of tuples that can be stored in the data chunk
	idx_t capacity;
```

We usually default-initialise it to `STANDARD_VECTOR_SIZE`. So we should ensure that `row` is within the data chunk's bounds, otherwise this can seg fault when trying to write the value to the data chunk's vector (let's also add a test).
```cpp
Vector::SetValue
...
	case PhysicalType::BOOL:
		reinterpret_cast<bool *>(data)[index] = val.GetValueUnsafe<bool>();
		break;
```

Reply:
My question was how to ensure that `row` is within the bounds to avoid seg fault, but there is no public method to get the capacity of the data chunk.